main.go

```diff
@@ -339,23 +255,61 @@ func parseAndRenderSegs(sourcePath string) []*Seg {
 // -----------------------------------------------------------------------------
 
 type exampleFile struct {
-	Segs []*Seg
+	// Lang is the language of code file, like `gop` / `go`
+	Lang string
+	// HeadDoc is document content before code content
+	HeadDoc []*Seg
+	// Code is all code segments of example file
+	Code []*Seg
+	// TailDoc is document content after code content
+	TailDoc []*Seg
 }
 
 type example struct {
 	*exampleIndex
 	Files []*exampleFile
 }
 
+// classifySegs classify segments into headDoc, code & tailDoc
+func classifySegs(segs []*Seg) (headDoc, code, tailDoc []*Seg) {
+	hasSeenCode := false
+	for _, seg := range segs {
+		if seg.Code != nil {
+			code = append(code, seg)
+			hasSeenCode = true
+			continue
+		}
+		if hasSeenCode {
+			// move all doc segs among code segs to tail,
+			// so that single code block will be rendered as result
+			tailDoc = append(tailDoc, seg)
```

Comment:
Here we assume that there is no doc-only segments among code segments; If there is, it will be rendered after all code segments - so that the code segments for a file will be rendered as a whole like this:

![image](https://user-images.githubusercontent.com/1492263/145243513-e35acc91-b46a-46a1-8fbd-151da677f10f.png)